sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/DeserializerBuildHelper.scala

```diff
@@ -48,8 +48,8 @@ object DeserializerBuildHelper {
       expr: Expression,
       dataType: DataType,
       nullable: Boolean,
-      walkedTypePath: Seq[String],
-      funcForCreatingNewExpr: (Expression, Seq[String]) => Expression): Expression = {
+      walkedTypePath: WalkedTypePath,
+      funcForCreatingNewExpr: (Expression, WalkedTypePath) => Expression): Expression = {
     val newExpr = funcForCreatingNewExpr(expr, walkedTypePath)
```

Comment:
this belongs to the previous PR: why not just let the caller side create the expression and pass to `deserializerForWithNullSafety`?

Reply:
Yeah I might be considered too complicated. Not a big deal and looks simpler. Will address. Thanks!